ecmascript/transforms/module/src/common_js.rs

```diff
@@ -84,7 +84,10 @@ where
 
         let mut exports = vec![];
         let mut initialized = IndexSet::default();
-        let mut export_alls = vec![];
+
+        let mut export_alls: AHashMap<JsWord, ModuleItem> = Default::default();
```

Comment:
I think storing `ModuleItem` in a hash map will be very slow, due to `memmove`.
`ModuleItem` is a big struct, so I think this case be performance issue.

Reply:
good point, was thinking that too but wasn't sure about actual implication so didn't try early optimizations. any suggestions?

Reply:
The reason I created moduleitem early is at the moment pushing stmt (https://github.com/swc-project/swc/pull/2598/files#diff-506530d02689c3c49a06cbec6fb8c9ffd7b1df4c551dcac29d8bd08d0248590fR712) there's mutable borrow to scope exists, so can't call `handle_export_all`. Curious if there's easy workaround.

Reply:
What you did is right.
Storing only minimum information to create `ModuleItem` and actually create big structs (`ModuleItem` in this case) works.